src/inmanta/deploy/scheduler.py

```diff
@@ -60,30 +115,38 @@ def __init__(
         self._work: work.ScheduledWork = work.ScheduledWork(
             requires=self._state.requires.requires_view(),
             provides=self._state.requires.provides_view(),
-            consumer_factory=self.start_for_agent,
+            new_agent_notify=self._start_for_agent,
```

Comment:
Just a suggestion: I found the name `consumer_factory` a bit vague / confusing. I can revert this if you don't like it.